pac4j-saml/src/main/java/org/pac4j/saml/client/SAML2ClientConfiguration.java

```diff
@@ -59,17 +69,45 @@ public SAML2ClientConfiguration() {}
     private String signatureCanonicalizationAlgorithm;
     private boolean wantsAssertionsSigned = true;
 
-    public SAML2ClientConfiguration(final String keystorePath, final String keystorePassword,
-                                    final String privateKeyPassword, final String identityProviderMetadataPath) {
-        this(keystorePath, keystorePassword, privateKeyPassword, identityProviderMetadataPath, null, null);
-    }
+	private String keyStoreAlias;
+
+	private String keyStoreType;
+
+	public SAML2ClientConfiguration(final KeyStore keystore, 
+			                        final String keyStoreAlias, 
+			                        final String keyStoreType,
+			                        final String privateKeyPassword, 
+			                        final Resource identityProviderMetadataResource) {
+		this(keystore, keyStoreAlias, keyStoreType, null, null, null, privateKeyPassword,
+				identityProviderMetadataResource, null, null, null);
+	}
+
+	public SAML2ClientConfiguration(final Resource keystoreResource, final String keyStoreAlias,
+			final String keyStoreType, final String keystorePassword, final String privateKeyPassword,
+			final Resource identityProviderMetadataResource) {
+		this(null, keyStoreAlias, keyStoreType, keystoreResource, null, keystorePassword, privateKeyPassword,
+				identityProviderMetadataResource, null, null, null);
+	}
 
     public SAML2ClientConfiguration(final String keystorePath, final String keystorePassword,
-                                    final String privateKeyPassword, final String identityProviderMetadataPath,
-                                    final String identityProviderEntityId, final String serviceProviderEntityId) {
+                                    final String privateKeyPassword, final String identityProviderMetadataPath) {
+        this(null, null, null, null, keystorePath, keystorePassword, privateKeyPassword, null,
+                identityProviderMetadataPath, null, null);
+    }
+
+	private SAML2ClientConfiguration(final KeyStore keyStore, final String keyStoreAlias, final String keyStoreType,
+			                         final Resource keystoreResource, final String keystorePath, final String keystorePassword,
```

Comment:
Mainly I wanted all your original code paths to still work so you could pass either the original keystorePath or the new keystoreResource. I figured someone in the wild may have made a call to the old methods and I didn't want my changes to break it. Is this not desirable?